ethers-providers/src/rpc/transports/common.rs

```diff
@@ -270,6 +271,101 @@ impl fmt::Display for Authorization {
     }
 }
 
+/// Default algorithm used for JWT token signing.
+const DEFAULT_ALGORITHM: Algorithm = Algorithm::HS256;
+
+/// JWT secret length in bytes.
+pub const JWT_SECRET_LENGTH: usize = 32;
+
+/// Generates a bearer token from a JWT secret
+pub struct JwtKey([u8; JWT_SECRET_LENGTH]);
+
+impl JwtKey {
+    /// Wrap given slice in `Self`. Returns an error if slice.len() != `JWT_SECRET_LENGTH`.
+    pub fn from_slice(key: &[u8]) -> Result<Self, String> {
+        if key.len() != JWT_SECRET_LENGTH {
+            return Err(format!(
+                "Invalid key length. Expected {} got {}",
+                JWT_SECRET_LENGTH,
+                key.len()
+            ))
+        }
+        let mut res = [0; JWT_SECRET_LENGTH];
+        res.copy_from_slice(key);
+        Ok(Self(res))
+    }
+
+    /// Decode the given string from hex (no 0x prefix), and attempt to create a key from it.
+    pub fn from_hex(hex: &str) -> Result<Self, String> {
+        let bytes = hex::decode(hex).map_err(|e| format!("Invalid hex: {}", e))?;
+        Self::from_slice(&bytes)
+    }
+
+    /// Returns a reference to the underlying byte array.
+    pub fn as_bytes(&self) -> &[u8] {
+        &self.0
+    }
```

Comment:
Needs a way to get the underlying bytes back

```suggestion
    /// Returns a reference to the underlying byte array.
    pub fn as_bytes(&self) -> &[u8; JWT_SECRET_LENGTH] {
        &self.0
    }
    
    pub fn into_bytes(self) ...
```

Reply:
makes sense, added